python/python/tests/test_pubsub.py

```diff
@@ -0,0 +1,1800 @@
+        )
+        # allow the messages to propagate
+        await asyncio.sleep(1)
+
+        async_msg = await listening_client.get_pubsub_message()
+        sync_msg = listening_client.try_get_pubsub_message()
+        assert sync_msg
+
+        assert async_msg.message == message
+        assert async_msg.channel == channel
+        assert async_msg.pattern is None
+
+        assert sync_msg.message == message2
+        assert sync_msg.channel == channel
+        assert sync_msg.pattern is None
+
+        # assert there are no messages to read
+        with pytest.raises(asyncio.TimeoutError):
+            await asyncio.wait_for(listening_client.get_pubsub_message(), timeout=3)
+
+        assert listening_client.try_get_pubsub_message() is None
+        if cluster_mode:
+            # Since all tests run on the same cluster, when closing the client, garbage collector can be called after another test will start running
+            # In cluster mode, we check how many subscriptions received the message
+            # So to avoid flakiness, we make sure to unsubscribe from the channels
+            await listening_client.custom_command(["SUNSUBSCRIBE", channel])
+
+    @pytest.mark.parametrize("cluster_mode", [True])
+    @pytest.mark.parametrize(
+        "method", [MethodTesting.Async, MethodTesting.Sync, MethodTesting.Callback]
+    )
+    async def test_sharded_pubsub_many_channels(
+        self, request, cluster_mode: bool, method: MethodTesting
+    ):
+        """
+        Test sharded Pub/Sub with multiple channels and different message retrieval methods.
+
+        This test verifies the behavior of sharded Pub/Sub when multiple messages are published
+        across multiple sharded channels. It covers three different message retrieval methods:
+        Async, Sync, and Callback.
+        """
+        NUM_CHANNELS = 256
+        shard_prefix = "{same-shard}"
+        publish_response = 1
+
+        # Create a map of channels to random messages with shard prefix
+        channels_and_messages = {
+            f"{shard_prefix}{get_random_string(10)}": get_random_string(5)
+            for _ in range(NUM_CHANNELS)
+        }
+
+        callback, context = None, None
+        callback_messages: List[
+            Union[RedisClient.PubSubMsg, RedisClusterClient.PubSubMsg]
+        ] = []
+        if method == MethodTesting.Callback:
+            callback = new_message
+            context = callback_messages
+
+        pub_sub = create_pubsub_subscription(
+            cluster_mode,
+            {PubSubChannelModes.Sharded: set(channels_and_messages.keys())},
+            callback=callback,
+            context=context,
+        )
+
+        publishing_client, listening_client = await setup_clients(
+            request, cluster_mode, pub_sub
+        )
+
+        min_version = "7.0.0"
+        if await check_if_server_version_lt(publishing_client, min_version):
+            pytest.skip(reason=f"Redis version required >= {min_version}")
```

Comment:
TIL: `spublish`/`ssubscribe` added in version 7.
How do you handle client creation with `ssubscribe` on redis 6, @ikolomi?
Is it tested, @shohamazon?

Reply:
Please add `Since` doc section to `SHARDED` subscription mode.